src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import '../../styles/transitions.scss';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null,
+  loading: boolean,
+  loadingTodoId: number[],
+  removeTodo: (id: number) => Promise<void>,
+  updateTodo: (id: number, data: string | boolean) => Promise<void>,
+  setErrorText: (errorText : string) => void,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos, tempTodo, loading, loadingTodoId, removeTodo, updateTodo, setErrorText,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      <TransitionGroup>
+        {todos.map(todo => {
+          return (
+            <CSSTransition
+              key={todo.id}
+              timeout={200}
+              classNames="item"
+            >
+              <TodoItem
+                key={todo.id}
+                todo={todo}
+                loading={false}
```

Comment:
The `loading` prop of the `TodoItem` should be dynamically set based on the `loadingTodoId` array. You can use `loadingTodoId.includes(todo.id)` to check if the current `todo.id` is being loaded.